### ISSUE TITLE
Can't Explicitly Add a Sitecore Wildcard Item to a Unicorn Config

### ISSUE THREAD

**Issue by jason-booth** [kamsar/Unicorn#142]
We're using the Sitecore wildcard module and want to explicitly add our wildcards to one of our Unicorn configs like:

```
<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
	<sitecore>
		<unicorn>
			<configurations>
				<configuration name="New Items Only" description="Syncs all items that are required for the code to work, but can be edited in the CMS. Only new items will be added/">
					<evaluator type="Unicorn.Evaluators.NewItemOnlyEvaluator, Unicorn" singleInstance="true"/>
					<dataProviderConfiguration enableTransparentSync="false" type="Unicorn.Data.DataProvider.DefaultUnicornDataProviderConfiguration, Unicorn" />
					<predicate type="Unicorn.Predicates.SerializationPresetPredicate, Unicorn" singleInstance="true">
						<include database="master" path="/sitecore/content/OurSite/Members/*"></include>
					</predicate>
				</configuration>
			</configurations>
		</unicorn>
	</sitecore>
</configuration>
```

We also want to exclude them from another config where they are being automatically included currently:

```
<configuration xmlns:patch="http://www.sitecore.net/xmlconfig/">
	<sitecore>
		<unicorn>
			<configurations>
				<configuration name="Content-CR-Members" description="Sitecore site's content">
					<predicate type="Unicorn.Predicates.SerializationPresetPredicate, Unicorn" singleInstance="true">
						<include database="master" path="/sitecore/content/Our Site/Members">
							<exclude path="/sitecore/content/Our Site/Members/*" />
						</include>
					</predicate>
			</configurations>
		</unicorn>
	</sitecore>
</configuration>
```

With this setup, we receive an error trying to load the Unicorn control panel:

```
[ArgumentException: Illegal characters in path.]
   System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional) +13750332
   System.Security.Permissions.FileIOPermission.QuickDemand(FileIOPermissionAccess access, String fullPath, Boolean checkForDuplicates, Boolean needFullPath) +161
   System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost) +55
   Rainbow.Storage.SerializationFileSystemTree..ctor(String name, String globalRootItemPath, String databaseName, String physicalRootPath, ISerializationFormatter formatter, Boolean useDataCache) +565
   Rainbow.Storage.SerializationFileSystemDataStore.CreateTree(TreeRoot root, ISerializationFormatter formatter, Boolean useDataCache) +148
   System.Linq.WhereSelectArrayIterator`2.MoveNext() +80
   System.Collections.Generic.List`1..ctor(IEnumerable`1 collection) +524
   System.Linq.Enumerable.ToList(IEnumerable`1 source) +70
   Rainbow.Storage.SerializationFileSystemDataStore..ctor(String physicalRootPath, Boolean useDataCache, ITreeRootFactory rootFactory, ISerializationFormatter formatter) +266
   lambda_method(Closure , Object[] ) +246
   Unicorn.Configuration.MicroConfiguration.Activate(Type type, KeyValuePair`2[] unmappedConstructorParameters) +825
   Unicorn.Configuration.<>c__DisplayClass7_0`1.<RegisterExpectedConfigType>b__2() +38
   System.Lazy`1.CreateValue() +244
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Lazy`1.get_Value() +14325153
   Unicorn.Data.ConfigurationDataStore.RegisterForChanges(Action`2 actionOnChange) +19
   Unicorn.Data.DataProvider.UnicornDataProvider..ctor(ITargetDataStore targetDataStore, ISourceDataStore sourceDataStore, IPredicate predicate, IFieldFilter fieldFilter, IUnicornDataProviderLogger logger, IUnicornDataProviderConfiguration dataProviderConfiguration, PredicateRootPathResolver rootPathResolver) +524
   lambda_method(Closure , Object[] ) +348
   Unicorn.Configuration.MicroConfiguration.Activate(Type type, KeyValuePair`2[] unmappedConstructorParameters) +825
   Unicorn.Configuration.MicroConfiguration.Resolve() +117
   System.Linq.WhereSelectArrayIterator`2.MoveNext() +80
   System.Linq.Buffer`1..ctor(IEnumerable`1 source) +250
   System.Linq.Enumerable.ToArray(IEnumerable`1 source) +106
   Unicorn.Data.DataProvider.UnicornSqlServerDataProvider..ctor(String connectionString) +232
```

Presumably it's trying to read a folder with a name of `*`?

When we don't try to explicitly include it in the first config and exclude it from the second, it is synced fine due to the second config including the wildcards' parent. Folders and yml files with the name `_` are created and all is happy. We need to move this into the first config though for some deployment considerations.

Is this a Unicorn issue or are we just doing the inclusion incorrectly?

Thanks!

**Comment by kamsar** [comment]
I'm guessing that the root item does not receive the same escaping semantics as the children.

**Comment by kamsar** [comment]
This will be fixed in the next release. There are several aspects to the fix:

1) The ability to use wildcard escaping to directly refer to wildcard items in the predicate by using `\*`: [example](https://github.com/kamsar/Unicorn/blob/master/src/Unicorn.Tests/Predicates/TestConfiguration.xml#L81)
2) Rainbow now validates that root paths are valid filesystem paths and throws a much more descriptive error. In your case, you'd been using `<include database="master" path="/sitecore/content/OurSite/Members/*"></include>`, which does not specify a `name` attribute. This means the default generated name is the last segment of the path - `*` - which is not a valid filesystem name. The new exception will tell you that you should explicitly specify a name attribute so as to give you a valid pathname (and a more descriptive folder name than say `_` which is also not good)

**Comment by kamsar** [comment]
Released in 3.3.1
